installer/app/src/views/azure-create-app-tutorial.js.jsx

```diff
@@ -5,163 +5,240 @@ import Sheet from './css/sheet';
+                </div>
+
+                {pagination.prev} {pagination.next} {pagination.submit}
+            </div>
+        );
+    },
+
+    componentDidMount: function () {
+        this.state.styleEl.commit();
+    },
+
+    /**
+     * Modifies the state's `tutorialSlide` variable, either advancing or stepping back in the
+     * tutorial's walkthrough.
+     * @param  {event} e
+     */
+    __handleAdvanceTutorialClick: function (e) {
+        var s = this.state;
+
+        // Validate Inputs
+        if (s.tutorialSlide === 5 && document.querySelector('[name=client_id]').value === '') {
+            document.querySelector('[name=client_id]').focus();
+            return;
+        } else if (s.tutorialSlide === 8 && document.querySelector('[name=endpoint]').value === '') {
+            document.querySelector('[name=endpoint]').focus();
+            return;
+        }
+
+        if (!s.tutorialSlide || s.tutorialSlide >= 9) {
+            s.tutorialSlide = null;
+        } else {
+            s.tutorialSlide = (e.target.id === 'azureTutNext') ? s.tutorialSlide + 1 : s.tutorialSlide - 1;
+        }
+
+        s.showRedirectURI = (s.tutorialSlide === 4);
+        s.showClientIDInput = (s.tutorialSlide === 5);
+        s.showEndpointInput = (s.tutorialSlide === 8);
+        s.done = (s.tutorialSlide === 9 || s.skipTutorial);
+
+        this.setState(s);
+    },
+
+    /**
+     * Select's the current's input contents for easier copying.
+     * @param  {event} e
+     */
+    __handleRedirectURIInputClick: function (e) {
+        InputSelection.selectAll(e.target);
+    },
+
+    /**
+     * Returns the styles the individual input fields, depending on whether or not they should
+     * be visible or not.
+     * @return {object} A on object with three keys (redirectURI, clientID, endpoint) with styles
+     */
+    __getInputStyles: function () {
+        var s = this.state,
+            redirectURI = (s.skipTutorial || s.showRedirectURI) ? {margin: '0 0 5px 0'} : {display: 'none', visibility: 'collapse'},
+            clientId = (s.skipTutorial || s.showClientIDInput) ? {margin: '0 0 5px 0'} : {display: 'none', visibility: 'collapse'},
+            endpoint = (s.skipTutorial || s.showEndpointInput) ? {margin: '0 0 5px 0'} : {display: 'none', visibility: 'collapse'};
+
+        return {
+            redirectURI: redirectURI,
+            clientId: clientId,
+            endpoint: endpoint
+        }
+    },
+
+    /**
+     * Skips the tutorial. The tutorial is already skipped and this method is called again, it
+     * resets the tutorial.
+     */
+    __skipTutorial: function () {
+        if (this.state.skipTutorial) {
+            return this.setState({skipTutorial: false});
+        } else {
+            this.setState({
+                skipTutorial: true,
+                tutorialSlide: undefined
```

Comment:
Use `null` and set an initial value in `getInitialState` (all state fields should be defined there).